src/game/common/rts/science.cpp

```diff
@@ -42,15 +43,13 @@ ScienceInfo::ScienceInfo() :
 
 void ScienceStore::Reset()
 {
-    for (auto it = m_infoVec.begin(); it != m_infoVec.end(); ++it) {
-        if (*it != nullptr) {
-            *it = reinterpret_cast<ScienceInfo *>((*it)->Delete_Overrides());
-        }
-
-        if (*it == nullptr) {
-            m_infoVec.erase(it);
+    // #BUGFIX Does no longer iterate with invalidated iterator.
+    rts::erase_if(m_infoVec, [](ScienceInfo *science) {
```

Comment:
and how does this call delete overrides on all of them...?

Reply:
I don't understand question. 2 lines below it calls `Delete_Overrides()`. It's a lambda passed to `erase_if` function that is called on each container element.